src/server/page_server.cpp

```diff
@@ -237,7 +203,115 @@ void page_server::on_data_page_read_result (const FILEIO_PAGE *io_page, int erro
 	}
     }
 
-  m_active_tran_server_conn->push (page_to_tran_request::SEND_DATA_PAGE, std::move (message));
+  m_conn->push (page_to_tran_request::SEND_DATA_PAGE, std::move (message));
+}
+
+void page_server::set_active_tran_server_connection (cubcomm::channel &&chn)
+{
+  assert (is_page_server ());
+
+  chn.set_channel_name ("ATS_PS_comm");
+  er_log_debug (ARG_FILE_LINE, "Active transaction server connected to this page server. Channel id: %s.\n",
+		chn.get_channel_id ().c_str ());
+
+  if (m_active_tran_server_conn != nullptr)
+    {
+      // ATS must have crashed without disconnecting from us. Destroy old connection to create a new one.
+      disconnect_tran_server (m_active_tran_server_conn.get ());
```

Comment:
See the suggestion in next comment.
```suggestion
      disconnect_active_tran_server ();
```